src/librustc_const_eval/eval.rs

```diff
@@ -1117,6 +1117,10 @@ fn cast_const<'tcx>(tcx: &TyCtxt<'tcx>, val: ConstVal, ty: ty::Ty) -> CastResult
         Float(f) => cast_const_float(tcx, f, ty),
         Char(c) => cast_const_int(tcx, Infer(c as u64), ty),
         Function(_) => Err(UnimplementedConstVal("casting fn pointers")),
+        ByteStr(_) => match ty.sty {
+            ty::TyRawPtr(_) => Err(ErrKind::UnimplementedConstVal("casting a bytestr to a raw ptr")),
```

Comment:
Yea, and the `const_err` lint ignores some kinds of errors, since there's no point in informing the user that const_eval can't evaluate it unless they try to use it in an array length or pattern.